src/irmin/tree_intf.ml

```diff
@@ -48,19 +48,19 @@ module type S = sig
   val of_node : node -> t
   (** [of_node n] is the subtree built from the node [n]. *)
 
-  val v : [< `Node of node | `Contents of contents * metadata ] -> t
+  type elt = [ `Node of node | `Contents of contents * metadata ]
+  (** The type for tree elements. *)
+
+  val v : elt -> t
   (** General-purpose constructor for trees. *)
 
-  val destruct : t -> [> `Node of node | `Contents of contents * metadata ]
+  val destruct : t -> elt
```

Comment:
```suggestion
  val destruct : t -> [> elt]
```

Reply:
Same here, I'm not sure we need/want this.